pkg/controller/instance/instance_controller.go

```diff
@@ -85,13 +88,35 @@ func (r *Reconciler) SetupWithManager(
 			return requests
 		})
 
+	hasAnnotation := func(key string, annotations map[string]string) bool {
+		if annotations == nil {
+			return false
+		}
+		for k := range annotations {
+			if k == key {
+				return true
+			}
+		}
+		return false
+	}
+
+	// resPredicate ignores DeleteEvents for pipe-pods only (marked with task.PipePodAnnotation)
```

Comment:
maybe you could write a bit about why we need to ignore it?

Reply:
What happens if this pod is deleted (e.g. by a user) while it's running? How do we handle this case during reconciliation when it is ignored here?

Reply:
> maybe you could write a bit about why we need to ignore it?

Good point. Done

> What happens if this pod is deleted (e.g. by a user) while it's running? 

A pipe-pod exists during a brief time when a pipe-task is running and is deleted afterward. Should a user manually delete it during this period, a plan might stay `IN_PROGRESS` indefinitely.